tests/e2e/suites.go

```diff
@@ -0,0 +1,288 @@
+		}
+		_, _, err := framework.RunCmd(cmd[0], cmd[1:]...)
+		framework.ExpectNoError(err, "failed to install kubernetes %s", kindName)
+	}
+
+	clientConfig, err := framework.LoadConfig()
+	framework.ExpectNoError(err, "failed load config")
+	runtimeClient, err := client.New(clientConfig, client.Options{})
+	framework.ExpectNoError(err)
+
+	err = waitForKubernetesWorkloadsReady("", 30*time.Minute, 10*time.Second, runtimeClient)
+	framework.ExpectNoError(err, "failed to wait for kubernetes workloads ready")
+}
+
+func setupCertManager() {
+	if !e2econfig.TestConfig.InstallCertManager {
+		return
+	}
+
+	helmName := "cert-manager" // nolint: goconst
+	helmNamespace := "cert-manager"
+	workloadNamespaces := []string{helmNamespace}
+	helmArgs := []string{
+		"install", helmName,
+		"cert-manager",
+		"--repo", "https://charts.jetstack.io",
+		"--namespace", helmNamespace,
+		"--create-namespace",
+		"--version", "v1.3.1",
+		"--set", "installCRDs=true",
+	}
+	helmInstall(helmName, helmNamespace, workloadNamespaces, helmArgs...)
+}
+
+func setupKruise() {
+	if !e2econfig.TestConfig.InstallKruise {
+		return
+	}
+	helmName := "kruise"
+	helmNamespace := "default"
+	workloadNamespaces := []string{"kruise-system"}
+	helmArgs := []string{
+		"install", helmName,
+		"https://github.com/openkruise/kruise/releases/download/v0.8.1/kruise-chart.tgz",
+	}
+	helmInstall(helmName, helmNamespace, workloadNamespaces, helmArgs...)
+}
+
+func setupNebulaOperator() {
+	if !e2econfig.TestConfig.InstallNebulaOperator {
+		return
+	}
+	helmName := "nebula-operator"
+	helmNamespace := "nebula-operator-system"
+	workloadNamespaces := []string{helmNamespace}
+	helmArgs := []string{
+		"install", helmName,
+		path.Join(framework.TestContext.RepoRoot, "charts/nebula-operator"),
+		"--namespace", helmNamespace,
+		"--create-namespace",
+	}
+	helmInstall(helmName, helmNamespace, workloadNamespaces, helmArgs...)
+}
+
+func cleanupKubernetes() {
+	if !e2econfig.TestConfig.InstallKubernetes || !e2econfig.TestConfig.UninstallKubernetes {
+		return
+	}
+	framework.Logf("uninstall kubernetes")
+
+	kindName := e2econfig.TestConfig.KindName
+	if kindName == "" {
+		kindName = e2econfig.DefaultKindName
+	}
+	_, _, err := framework.RunCmd("kind", "delete", "cluster", "--name", kindName)
+	framework.ExpectNoError(err, "failed to uninstall kubernetes")
+}
+
+func isHelmInstalled(name, namespace string) bool {
+	args := []string{"status", name, "--namespace", namespace}
+
+	if framework.TestContext.KubeConfig != "" {
+		args = append(args, "--kubeconfig", framework.TestContext.KubeConfig)
+	}
+
+	_, _, err := framework.RunCmd("helm", args...)
+	if err != nil {
+		if !strings.Contains(err.Error(), "release: not found") {
+			framework.ExpectNoError(err, "failed to helm status %s%s", namespace, name)
+		}
+		return false
+	}
+	return true
+}
+
+func helmInstall(name, namespace string, workloadNamespaces []string, args ...string) {
```

Comment:
I think install helm charts [kruise,cert-manager,nebula-operator] by e2e.sh shell scripts  more concise,  we can check if they installed or not in e2e framework

Reply:
The shell script should be executed before the test starts. And there may be different behaviors between each test case. With `go`, we can have better control.